File: test/src/main/java/org/sqlunet/browser/Actions.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.browser;

import android.view.View;

import org.hamcrest.Matcher;

import androidx.annotation.NonNull;
import androidx.test.espresso.UiController;
import androidx.test.espresso.ViewAction;

@SuppressWarnings("WeakerAccess")
public class Actions
{
	@NonNull
	static ViewAction onlyIf(@NonNull final ViewAction action, @NonNull final Matcher<View> constraints)
	{
		return new ViewAction()
		{
			@NonNull
			@Override
			public Matcher<View> getConstraints()
			{
				return constraints;
			}

			@Override
			public String getDescription()
			{
				return action.getDescription();
			}

			@Override
			public void perform(UiController uiController, View view)
			{
				action.perform(uiController, view);
			}
		};
	}
}

File: test/src/main/java/org/sqlunet/browser/Actions.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.browser

import android.view.View
import androidx.test.espresso.UiController
import androidx.test.espresso.ViewAction
import org.hamcrest.Matcher

object Actions {

    @JvmStatic
    fun onlyIf(action: ViewAction, constraints: Matcher<View>): ViewAction {

        return object : ViewAction {
            override fun getConstraints(): Matcher<View> {
                return constraints
            }

            override fun getDescription(): String {
                return action.description
            }

            override fun perform(uiController: UiController, view: View) {
                action.perform(uiController, view)
            }
        }
    }
}
